src/handler/http/models/destinations.rs

```diff
@@ -0,0 +1,241 @@
+        if let Some(template) = self.template {
+            let destination_type = match self.destination_type {
+                DestinationType::Email => meta_dest::DestinationType::Email(meta_dest::Email {
+                    recipients: self.emails,
+                }),
+                DestinationType::Http => meta_dest::DestinationType::Http(meta_dest::Endpoint {
+                    url: self.url,
+                    method: self.method,
+                    skip_tls_verify: self.skip_tls_verify,
+                    headers: self.headers,
+                }),
+                DestinationType::Sns => meta_dest::DestinationType::Sns(meta_dest::AwsSns {
+                    sns_topic_arn: self.sns_topic_arn.ok_or(DestinationError::InvalidSns)?,
+                    aws_region: self.aws_region.ok_or(DestinationError::InvalidSns)?,
+                }),
+            };
+            Ok(meta_dest::Destination {
+                id,
+                org_id: org_id.to_string(),
+                name: self.name,
+                module: meta_dest::Module::Alert {
+                    template,
+                    destination_type,
+                },
+            })
+        } else if let Some(pipeline_id) = self.pipeline_id {
+            let endpoint = meta_dest::Endpoint {
+                url: self.url,
+                method: self.method,
+                skip_tls_verify: self.skip_tls_verify,
+                headers: self.headers,
+            };
+            Ok(meta_dest::Destination {
+                id,
+                org_id: org_id.to_string(),
+                name: self.name,
+                module: meta_dest::Module::Pipeline {
+                    pipeline_id,
+                    endpoint,
+                },
+            })
+        } else {
+            Err(DestinationError::UnsupportedType)
+        }
+    }
+}
+
+impl From<meta_dest::Template> for Template {
+    fn from(value: meta_dest::Template) -> Self {
+        let (title, template_type) = match value.template_type {
+            meta_dest::TemplateType::Email { title } => (title, DestinationType::Email),
+            meta_dest::TemplateType::Http => (String::new(), DestinationType::Http),
+            meta_dest::TemplateType::Sns => (String::new(), DestinationType::Sns),
+        };
+
+        Self {
+            name: value.name,
+            body: value.body,
+            is_default: value.is_default.then_some(true),
+            template_type,
+            title,
+        }
+    }
+}
+
+impl Template {
+    pub fn into(self, org_id: &str) -> meta_dest::Template {
+        let template_type = match self.template_type {
+            DestinationType::Email => meta_dest::TemplateType::Email { title: self.title },
+            DestinationType::Sns => meta_dest::TemplateType::Sns,
+            DestinationType::Http => meta_dest::TemplateType::Http,
+        };
+        meta_dest::Template {
+            id: ider::uuid(),
```

Comment:
I'd be careful about creating the KSUID here and would instead try to restrict KSUID generation just to the DB Layer.

One potential issue I can see with generating the KSUID here is that if someone were to implement a "DB put" method that accepts a `meta_det::Template` parameter, then the "DB put" method will first try to look up an existing record with the given KSUID. If the KSUID is assigned here, then when the DB returns back no result it is ambiguous if either...
- the record with the given KSUID never existed and never was expected to exist, so it should be created
- or if the record with the given KSUID was actually expected to exist but could not be found, so an error should be returned rather than creating a new record